handler/src/main/java/io/netty/handler/ipfilter/IpSubnetFilterRule.java

```diff
@@ -68,20 +68,35 @@ public IpFilterRuleType ruleType() {
         return filterRule.ruleType();
     }
 
+    @Override
+    public int compareTo(IpSubnetFilterRule o) {
+        return getNetworkAddress().compareTo(o.getNetworkAddress());
+    }
+
+    private BigInteger getNetworkAddress() {
+        if (filterRule instanceof Ip4SubnetFilterRule) {
+            return ((Ip4SubnetFilterRule) filterRule).networkAddressAsBigInteger;
+        } else {
+            return ((Ip6SubnetFilterRule) filterRule).networkAddress;
+        }
+    }
+
     private static final class Ip4SubnetFilterRule implements IpFilterRule {
 
+        private final BigInteger networkAddressAsBigInteger;
```

Comment:
I wonder if we can implement a comparison without needing to keep this around, or if that adds too much overhead.

Reply:
Yeah, I'll use Integer directly here now.